src/main/java/org/ow2/proactive/catalog/rest/exceptiohandler/ExceptionHandlerController.java

```diff
@@ -28,28 +28,43 @@
 import javax.servlet.http.HttpServletRequest;
 
 import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
+import lombok.extern.log4j.Log4j;
+
 
 /**
  * @author ActiveEon Team
  * @since 03/08/2017
  */
 @Controller
 @ControllerAdvice
+@Log4j
 public class ExceptionHandlerController {
 
     @ExceptionHandler(Exception.class)
-    public ResponseEntity exceptionHandler(HttpServletRequest request, Exception exception) {
+    @ResponseBody
+    public ResponseEntity exceptionHandler(HttpServletRequest request, Exception exception) throws Exception {
+        log.warn("Exception: " + exception.getLocalizedMessage());
 
-        ResponseStatus responseStatusAnnotation = AnnotationUtils.findAnnotation(exception.getClass(),
-                                                                                 ResponseStatus.class);
-        return new ResponseEntity(responseStatusAnnotation.code());
+        HttpStatus responseStatusCode = resolveAnnotatedResponseStatus(exception);
 
+        return ResponseEntity.status(responseStatusCode)
+                             .body(new ExceptionRepresentation(responseStatusCode.value(),
+                                                               exception.getLocalizedMessage()));
     }
 
+    HttpStatus resolveAnnotatedResponseStatus(Exception exception) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)